qir/qat/Rules/Factory.cpp.orig

```diff
@@ -0,0 +1,801 @@
+
+        if (config.optimizeResultZero())
+        {
+            optimizeResultZero();
+        }
+
+        if (config.useStaticQubitArrayAllocation())
+        {
+            useStaticQubitArrayAllocation();
+        }
+
+        if (config.useStaticQubitAllocation())
+        {
+            useStaticQubitAllocation();
+        }
+
+        if (config.useStaticResultAllocation())
+        {
+            useStaticResultAllocation();
+        }
+
+        if (config.removeGetZeroOrOne())
+        {
+            removeGetZeroOrOne();
+        }
+    }
+
+    void RuleFactory::removeFunctionCall(String const& name)
+    {
+        addRule({callByNameOnly(name), deleteInstruction()});
+    }
+
+    void RuleFactory::removeGetZeroOrOne()
+    {
+        addRule(
+            {callByNameOnly("__quantum__rt__result_get_zero"), deleteUnusedInstruction()},
+            RuleSet::ReplaceDirection::ReplaceBackwards);
+        addRule(
+            {callByNameOnly("__quantum__rt__result_get_one"), deleteUnusedInstruction()},
+            RuleSet::ReplaceDirection::ReplaceBackwards);
+    }
+
+    void RuleFactory::resolveConstantArraySizes()
+    {
+        /// Array access replacement
+        auto size_replacer = [](Builder&, Value* val, Captures& cap, Replacements& replacements)
+        {
+            // Get the index and testing that it is a constant int
+            auto cst = llvm::dyn_cast<llvm::ConstantInt>(cap["size"]);
+            if (cst == nullptr)
+            {
+                // ... if not, we cannot perform the mapping.
+                return false;
+            }
+
+            val->replaceAllUsesWith(cst);
+            replacements.push_back({llvm::dyn_cast<Instruction>(val), nullptr});
+
+            return true;
+        };
+
+        auto create_array = call("__quantum__rt__array_create_1d", "elementSize"_cap = _, "size"_cap = _);
+        auto get_size     = call("__quantum__rt__array_get_size_1d", create_array);
+
+        addRule({std::move(get_size), size_replacer});
+    }
+
+    void RuleFactory::inlineCallables()
+    {
+        auto logger = logger_;
+
+        /// Array access replacement
+<<<<<<< HEAD
+        auto callable_replacer = [](Builder&, Value* val, Captures& captures, Replacements&)
+        {
+            llvm::errs() << "FOUND CALLABLE\n";
+            llvm::errs() << *val << "\n";
+            llvm::errs() << "Calling " << *captures["function"] << "\n";
+=======
+        auto callable_replacer = [logger](Builder&, Value* val, Captures&, Replacements&) {
+            if (logger)
+            {
+                logger->setLocationFromValue(val);
+                logger->internalError("Support for callable replacement is not implemented yet.");
+            }
+            else
+            {
+                throw std::runtime_error(
+                    "No logger present - internal error: Support for callable replacement is not implemented "
+                    "yet.\n");
+            }
+
+>>>>>>> main
```

Comment:
L143 - 163 look like an unresolved conflict. This is strange for a newly created file. 
What is this file for? Did you forget to remove it?

Reply:
Multiple occurrences.

Reply:
Multiple `.orig` files.

Reply:
All removed. Thanks!